build-logic-commons/build-scan/src/main/kotlin/gradlebuild.cache-miss-monitor.gradle.kts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.gradle.scan.plugin.BuildScanExtension
+import java.util.concurrent.atomic.AtomicBoolean
+
+val cacheMissTagged = AtomicBoolean(false)
+val buildScan = extensions.findByType<BuildScanExtension>()
```

Comment:
I think it would be better if this plugin reacts to the build scan plugin being applied, instead of assuming that it has been applied.